### PR TITLE
Docs restructure and improvement for v3

### PR DESCRIPTION
**Pull request by SGevorg**

### CODE REVIEW

**Comment by SGevorg** [comment]
Next step:
- Add collab links and setup public collab-generated content example

**Comment by SGevorg** [comment]
- Collab added
Next Step:
- Setup full tutorial infra inspired by https://devblog.pytorchlightning.ai/publishing-lightning-tutorials-cbea3eaa4b2c

File: aim/storage/encoding/encoding_native.cpp [hunk 1]
```diff
@@ -0,0 +1,5509 @@
+/* Generated by Cython 0.29.24 */
```
Comment (mahnerak) [inline aim/storage/encoding/encoding_native.cpp:1]: The auto-generated sources should not appear in the git repo.

**Review by mahnerak — CHANGES_REQUESTED** [review]

File: aim/storage/encoding/encoding_native.cpp (hunk 1 above)
Comment (SGevorg) [reply aim/storage/encoding/encoding_native.cpp:1]: yea, will remove them, I have moved away from this solution alltogether.
will push the Pytorch Lightning  sphinx based one soon

**Comment by SGevorg** [comment]
Basically in the last two commits:
- added full initial structure of the docs with markdowns
- filled in wherever I could from the current readme/docs

File: aim/storage/encoding/encoding_native.cpp (hunk 1 above)
Comment (SGevorg) [reply aim/storage/encoding/encoding_native.cpp]: @mahnerak removed the autogenerated files

File: docs/source/aim_sdk.rst [hunk 2]
```diff
@@ -1,16 +1,6 @@
 aim.sdk subpackage
```
Comment (alberttorosyan) [inline docs/source/aim_sdk.rst:1]: These files are generated by Sphinx when running `sphinx apidoc` command. The naming follows python modules. Maybe we should keep that to not have conflicts when re-generating docs skeleton?

File: docs/source/conf.py [hunk 3]
```diff
@@ -12,18 +12,55 @@
...
 sys.path.insert(0, os.path.abspath('..'))
 
 
+PATH_HERE = os.path.abspath(os.path.dirname(__file__))
+PATH_ROOT = os.path.join(PATH_HERE, "..", "..")
+
+FOLDER_GENERATED = "generated"
+
+spec = spec_from_file_location(
+    "aim/__about__.py", os.path.join(PATH_ROOT, "aim", "__about__.py")
```
Comment (alberttorosyan) [inline docs/source/conf.py:27]: Lets stick to snigle-quotes please 🙏.

File: docs/source/md/cli_references.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+Aim CLI offers a simple interface to easily organize and record your experiments.
+Paired with the [Python Library](#sdk-specifications), Aim is a powerful utility to record, search and compare AI experiments.
+Here are the set of commands supported:
+
+
+| Command       | Description                                                          |
+| --------------| -------------------------------------------------------------------- |
+| `init`        | Initialize the `aim` repository.                                     |
+| `version`     | Displays the version of aim cli currently installed.                 |
+| `experiment`  | Creates a new experiment to group similar training runs into.        |
```
Comment (alberttorosyan) [inline docs/source/md/cli_references.md:12]: `aim experiment` command had been removed.

File: docs/source/md/cli_references.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  **_Beware:_** Re-initialization of the repo clears `.aim` folder from previously saved data and initializes new repo.
+  **_Note:_** This command is not necessary to be able to get started with Aim as aim is automatically initializes with the first aim function call.
+
+### version
+Display the Aim version installed.
+```shell
+$ aim version
+```
+
+### experiment
```
Comment (alberttorosyan) [inline docs/source/md/cli_references.md:33]: same here; we don't have `aim experiment` anymore

File: docs/source/md/cli_references.md (hunk 4 above)
Comment (alberttorosyan) [reply docs/source/md/cli_references.md:12]: also need to add `aim upgrade` command here. It's used to convert legacy repos to the v3.0.0 format (and will be used for future backwards-incompatible changes as well)

File: docs/source/md/cli_references.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+| Args                              | Description                                               |
+| --------------------------------- | --------------------------------------------------------- |
+| `-h` &#124; `--host <host>`       | Specify host address.                                     |
+| `-p` &#124; `--port <port>`       | Specify port to listen to.                                |
+| `--repo <repo_path>`              | Path to parent directory of `.aim` repo. _Current working directory by default_ |
+| `--tf_logs <logs_dir_path>`       | Use Aim to search and compare TensorBoard experiments. More details in [TensorBoard Experiments](#tensorboard-experiments) |
+| `--dev`                           | Run UI in development mode.                                   |
+
+***Please make sure to run `aim up` in the directory where `.aim` is located.***
```
Comment (alberttorosyan) [inline docs/source/md/cli_references.md:61]: I guess this is not a limitation anymore? We have the `--repo` argument so the command can be executed from any directory.

File: docs/source/md/quick-start/integrations.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Easily integrate Aim with your favorite framework / tool
+
+### Python script
+
+```python
+import aim
+
+# Save inputs, hparams or any other `key: value` pairs
+aim.set_params(hyperparam_dict, name='hparams') # Passing name argument is optional
```
Comment (alberttorosyan) [inline docs/source/md/quick-start/integrations.md:11]: Lets use the current API for this. The top-level functions, such as `set_params`, `track`, `select` are kept for the sake of beckrads compatibility. Not sure we want to even document them.

File: docs/source/md/quick-start/why_aim.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+## Why do you need Aim?
+
+
+### Aim vs Tensorboard
```
Comment (alberttorosyan) [inline docs/source/md/quick-start/why_aim.md:4]: Do we want to do this in docs? Maybe better to leave the comparison for blog-posts?

**Review by alberttorosyan — CHANGES_REQUESTED** [review]

File: docs/source/aim_sdk.rst (hunk 2 above)
Comment (SGevorg) [reply docs/source/aim_sdk.rst:1]: hmm, does it generate for the api? 
maybe. we could generate in a separate folder as well.

File: docs/source/md/cli_references.md (hunk 5 above)
Comment (SGevorg) [reply docs/source/md/cli_references.md:33]: thnks, will fix this.

Comment (SGevorg) [reply docs/source/md/cli_references.md:33]: let's do another iteration and go through this afterwards ?

File: docs/source/conf.py (hunk 3 above)
Comment (SGevorg) [reply docs/source/conf.py:27]: will fix it thanks

File: docs/source/md/cli_references.md (hunk 5 above)
Comment (alberttorosyan) [reply docs/source/md/cli_references.md:33]: resolving since we have a task for CLI section

File: docs/source/aim_sdk.rst (hunk 2 above)
Comment (alberttorosyan) [reply docs/source/aim_sdk.rst:1]: I mean, the whole `source` directory in docs/ is the sphinx stuff. We still do some tweaks (for example, skip some source-code directories for now). there, but I think we should keep the file naming as it is.

File: docs/source/md/quick-start/integrations.md (hunk 7 above)
Comment (alberttorosyan) [reply docs/source/md/quick-start/integrations.md:11]: dropping this comment as well. we have a task for API reference tidy-up.

File: docs/source/aim_sdk.rst (hunk 2 above)
Comment (SGevorg) [reply docs/source/aim_sdk.rst:1]: yea, but in other places I saw (such as PL) the source was the whole docs. I doubt all of it was generated. moreover, in PL they had handpicked classes in the API reference.

File: docs/source/md/quick-start/why_aim.md (hunk 8 above)
Comment (SGevorg) [reply docs/source/md/quick-start/why_aim.md:4]: We have had number of requests for this. We need it everywhere lol

Comment (alberttorosyan) [reply docs/source/md/quick-start/why_aim.md:4]: haha, good point :)

File: docs/source/aim_sdk.rst (hunk 2 above)
Comment (alberttorosyan) [reply docs/source/aim_sdk.rst:1]: ok, lets see how it goes in a long run.

**Review by alberttorosyan — APPROVED** [review]

**Review by mahnerak — APPROVED** [review]

**Pull request merged**